File: csharp/dataRow.cs

using System.Collections.Generic;


public class dataRow
{
  public string id { get; set; }

  public float X { get; set; }

  public float Y { get; set; }

  public float pickupTime { get; set; }

  public float dropoffTime { get; set; }

  public float quanity { get; set; }

  public static List<dataRow> LoadData(string filename, int maxItems = int.MaxValue)
  {
    List<dataRow> res = new List<dataRow>();
    var lines = System.IO.File.ReadAllLines(filename);
    bool ns3format = (lines[0].Split(",")[0] == "id" && lines[0].Split(",")[1] == "name");
    for (int i = 1; i < lines.Length; i++)
    {
      if (res.Count >= maxItems)
      {
        break;
      }
      var line = lines[i];
      var items = line.Split(",");
      if (items.Length < 4)
      {
        // just the locations and geocodes
        res.Add(new dataRow
        {
          id = items[0],
          X = float.Parse(items[1], System.Globalization.CultureInfo.InvariantCulture),
          Y = float.Parse(items[2], System.Globalization.CultureInfo.InvariantCulture)
        });
      }
      else
      {
        if(ns3format){
          res.Add(new dataRow
          {
            id = items[0],
            X = float.Parse(items[2], System.Globalization.CultureInfo.InvariantCulture),
            Y = float.Parse(items[3], System.Globalization.CultureInfo.InvariantCulture),
            quanity = float.Parse(items[4], System.Globalization.CultureInfo.InvariantCulture)
          });
        }else{
          // load the pickup/dropoff times and quantites
          res.Add(new dataRow
          {
            id = items[0],
            X = float.Parse(items[1], System.Globalization.CultureInfo.InvariantCulture),
            Y = float.Parse(items[2], System.Globalization.CultureInfo.InvariantCulture),
            pickupTime = float.Parse(items[3], System.Globalization.CultureInfo.InvariantCulture),
            dropoffTime = float.Parse(items[4], System.Globalization.CultureInfo.InvariantCulture),
            quanity = float.Parse(items[5], System.Globalization.CultureInfo.InvariantCulture)
          });
        }        
      }
    }
    return res;
  }
}

File: java/src/main/java/icepackai/dataRow.java
package icepackai;

import java.util.*;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.charset.Charset;

public class dataRow {
  public dataRow(String _id, float x, float y) {
    id = _id;
    X = x;
    Y = y;
  };

  public dataRow(String _id, float x, float y, float ptime, float dtime, float qty) {
    id = _id;
    X = x;
    Y = y;
    pickupTime = ptime;
    dropoffTime = dtime;
    quantity = qty;
  };

  public String id;
  public float X;
  public float Y;
  public float pickupTime;
  public float dropoffTime;
  public float quantity;

  public static List<dataRow> LoadData(String filename) throws Exception {
    List<dataRow> res = new ArrayList<dataRow>();
    List<String> lines = Files.readAllLines(Paths.get(filename), Charset.defaultCharset());

    Boolean ns3format =
        (lines.get(0).split(",")[0].equals("id") && lines.get(0).split(",")[1].equals("name"));

    for (int i = 1; i < lines.size(); i++) {
      String line = lines.get(i);
      String[] items = line.split(",");
      if (items.length < 4) {
        res.add(new dataRow(items[0], Float.parseFloat(items[1]), Float.parseFloat(items[2])));
      } else {
        if (ns3format) {
          res.add(new dataRow(items[0], Float.parseFloat(items[2]), Float.parseFloat(items[3]), 0,
              0, Float.parseFloat(items[4])));
        } else {
          res.add(new dataRow(items[0], Float.parseFloat(items[1]), Float.parseFloat(items[2]),
              Float.parseFloat(items[3]), Float.parseFloat(items[4]), Float.parseFloat(items[5])));
        }
      }
    }
    return res;
  }
};
